networkx/classes/tests/test_digraph.py

```diff
@@ -244,6 +244,9 @@ def test_add_edge(self):
         assert G.adj == {0: {1: {}}, 1: {}}
         assert G.succ == {0: {1: {}}, 1: {}}
         assert G.pred == {0: {}, 1: {0: {}}}
+        G = self.Graph()
+        with pytest.raises(ValueError):
+            G.add_edge(None, 3)
```

Comment:
Just a minor nit, but there's no need to create a new graph here - might as well use the instance that already exists!

Also, we can improve the specificity of the test to make sure that the *correct* exception is being raised by checking the exception message. `pytest` provides a very nice way to do so with the `match=` kwarg:

```suggestion
        with pytest.raises(ValueError, match="None cannot be a node"):
            G.add_edge(None, 3)
```